common/telemetry/exporter/otlp_grpc.go

```
@@ -0,0 +1,40 @@
+		logger.Infof("Using secure gRPC connection for OTLP %s exporter", signalType)
+		transportCreds = credentials.NewTLS(&tls.Config{})
+	}
+
+	dialOpts := []grpc.DialOption{
+		grpc.WithTransportCredentials(transportCreds),
+	}
+
+	dialCtx, cancel := context.WithTimeout(ctx, cfg.OtelExporterOtlpTimeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(dialCtx, cfg.OtelExporterOtlpEndpoint, dialOpts...)
```

golangci-lint (1.64.8)

SA1019: grpc.DialContext is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)